### PR TITLE
Undoing a layer rename reverts to rename when going back to edit mode.

### LINKED ISSUE #11686: Undoing a layer rename reverts to rename when going back to edit mode.
## Bug Description

After renaming a layer and undoing the layer name the layer name uses the wrong value when re-entering edit mode.

See:
https://github.com/GoogleForCreators/web-stories-wp/pull/11676#issuecomment-1153807461

## Expected Behaviour

When editing the layer name it contains the current value for the layer.

## Steps to Reproduce

- Add a layer (layer has default name)
- Rename a layer ('my layer')
- Undo the rename (layer has default name)
- 2x click to rename again --- editable text now shows previous rename ('my layer')

## Additional Context

- Plugin Version:
- WordPress Version:
- Operating System:
- Browser:

### PR DESCRIPTION
**Pull request by timarney**
## Context

After renaming a layer and undoing the layer name the layer name uses the wrong value when re-entering edit mode.

See:
https://github.com/GoogleForCreators/web-stories-wp/pull/11676#issuecomment-1153807461

## Summary

I think the issue here is the useState + initial value

> the initial value matters only the first time a component renders. After that, it will preserve the state between renders.

https://dev.to/cronokirby/react-pitfalls-usestate-initialization-39be

This PR adds a useEffect to ensure the "initial value" element.layerName gets updates

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:
1. Enable the layer renaming feature
2. Add a new element to the stage
3. Open the layers panel
4. Rename the layer
5. Undo the rename 
6. Enter edit mode to rename the layer again --- the value should match the current layer name

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #11686

### CODE REVIEW

File: packages/story-editor/src/components/panels/design/layer/layer.js [hunk 1]
```diff
@@ -352,6 +352,10 @@ function Layer({ element }) {
     })
   );
 
+  useEffect(() => {
+    setNewLayerName(layerName);
+  }, [layerName]);
```
Comment (barklund) [inline packages/story-editor/src/components/panels/design/layer/layer.js:357]: With this, I think we can also remove  almost all the other places, where `setNewLayerName` is invoked, as most of them are workarounds to exactly this thing missing.

**Review by barklund — APPROVED** [review]
Works great!

File: packages/story-editor/src/components/panels/design/layer/layer.js (hunk 1 above)
Comment (timarney) [reply packages/story-editor/src/components/panels/design/layer/layer.js:357]: Okay so 

I think we'll still need the `handleChange` and `handleKeyDown`

But we can update `updateLayerName` to be this code below ya?

```javascript
const updateLayerName = () => {
    setRenamableLayer(null);
    const trimmedLayerName = newLayerName.trim();
    if (trimmedLayerName) {
      updateElementById({
        elementId: element.id,
        properties: { layerName: trimmedLayerName },
      });
    }
  };
```

Comment (timarney) [reply packages/story-editor/src/components/panels/design/layer/layer.js:357]: Pushed an update for that 

https://github.com/GoogleForCreators/web-stories-wp/pull/11700/commits/c1680cdb393ce3094b03553aed31499fb3db82d3

**Comment by merapi** [comment]
Oh, this should go to QA first, but since it's a tiny fix and all of us 3 already tested it - I suggest leaving/accepting it as is.

**Pull request merged**